core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/SearchIntentProcessing.kt

```diff
@@ -36,6 +38,11 @@ data class SearchIntentProcessing(
   @TargetApi(VERSION_CODES.M)
   override fun invokeWith(activity: AppCompatActivity) {
     if (intent != null) {
+      if (intent.getBooleanExtra(Constants.TAG_FROM_TAB_SWITCHER, false)) {
```

Comment:
I'd move this into the `offer(ScreenOrigin( if(fromTab) FromTabView else FromWebView)))`